### ISSUE TITLE
function_left_parentheses[_linter] discrepancy

### ISSUE THREAD

**Issue by russHyde** [jimhester/lintr#426]
I ran  the script in the readme (https://github.com/jimhester/lintr#project-configuration) to make a `.lintr` config for `installr` (https://github.com/talgalili/installr).
This gave the following .lintr:
```
linters: with_defaults(
    trailing_whitespace_linter = NULL, # 983
    line_length_linter = NULL, # 807
    object_name_linter = NULL, # 309
    commas_linter = NULL, # 253
    commented_code_linter = NULL, # 214
    spaces_left_parentheses_linter = NULL, # 176
    infix_spaces_linter = NULL, # 134
    no_tab_linter = NULL, # 102
    open_curly_linter = NULL, # 58
    paren_brace_linter = NULL, # 36
    single_quotes_linter = NULL, # 34
    trailing_blank_lines_linter = NULL, # 34
    spaces_inside_linter = NULL, # 21
    assignment_linter = NULL, # 19
    object_usage_linter = NULL, # 18
    object_length_linter = NULL, # 16
    closed_curly_linter = NULL, # 8
    function_left_parentheses = NULL, # 7
    cyclocomp_linter = NULL, # 2
    seq_linter = NULL, # 1
    dummy_linter = NULL
  )
exclusions: list(
    ...
)
```

Then, without modifying the .lintr file, I ran `lint_package()` on the same package.

This flagged up 7 lint failures  of the form "Remove spaces before the left parenthesis in a function call."

This seemed a bit weird, since all default linters have been NULLed out in the .lintr file. It appears to relate to a mismatch between the function-name `function_left_parentheses_linter` and the stated-name of the linter within that function `function_left_parentheses`.

```
          Lint(
            filename = source_file$filename,
            line_number = parsed$line1,
            column_number = parsed$col1,
            type = "style",
            message = "Remove spaces before the left parenthesis in a function call.",
            line = line,
            linter = "function_left_parentheses" # --- here
            )
```

`default_linters` uses the function-name to distinguish linters;
The results from as.data.frame(lint_package()) use the `$linter` entry returned by a call to `Lint(...)`.

Should the `Lint( ... linter = "", ...)` entry be equal to the name of the linter function for each linter?

I ran this to find any `linter = "..."` strings where the string did not end with `_linter`
```
# in lintr directory
grep "linter = " R/*.R | grep -v "_linter\""
```

... and checked the idenitified files, the only linters where the function-name and the linter-name in Lint() are mismatched are in `function_left_parentheses_linter` and in `get_source_expressions` (where an error string is returned).

**Comment by russHyde** [comment]
closed in #427
